presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -158,6 +158,42 @@ public void testParsingError()
         assertQueryFails("SELECT foo FROM", "line 1:16: mismatched input '<EOF>'. Expecting: .*");
     }
 
+    @Test
+    public void testEmptyTableName()
+    {
+        assertQueryFails("SELECT * FROM \"\"", "line 1:15: Table name is empty");
+    }
+
+    @Test
+    public void testNullTableName()
+    {
+        assertQueryFails("SELECT * FROM NULL", "line 1:15: mismatched input 'NULL'. Expecting: .*");
+    }
+
+    @Test
+    public void testEmptySchemaName()
+    {
+        assertQueryFails("SELECT * FROM \"\".foo", "line 1:15: Schema name is empty");
+    }
+
+    @Test
+    public void testNullSchemaName()
+    {
+        assertQueryFails("SELECT * FROM NULL.foo", "line 1:15: mismatched input 'NULL'. Expecting: .*");
+    }
+
+    @Test
+    public void testEmptyCatalogName()
+    {
+        assertQueryFails("SELECT * FROM \"\".schema.foo", "line 1:15: Catalog  does not exist");
+    }
+
+    @Test
+    public void testNullCatalogName()
+    {
+        assertQueryFails("SELECT * FROM NULL.schema.foo", "line 1:15: mismatched input 'NULL'. Expecting: .*");
```

Comment:
There is no need to run these tests per each connector. Add them somewhere else than ATQ.

Reply:
Good point, missed that!